homeassistant/components/vallox/binary_sensor.py

```diff
@@ -0,0 +1,87 @@
+"""Support for Vallox ventilation unit binary sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import ValloxDataUpdateCoordinator
+from .const import DOMAIN
+
+
+class ValloxBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    """Representation of a Vallox binary sensor."""
+
+    entity_description: ValloxBinarySensorEntityDescription
+    coordinator: ValloxDataUpdateCoordinator
+
+    def __init__(
+        self,
+        name: str,
+        coordinator: ValloxDataUpdateCoordinator,
+        description: ValloxBinarySensorEntityDescription,
+    ) -> None:
+        """Initialize the Vallox binary sensor."""
+        super().__init__(coordinator)
+
+        self.entity_description = description
+
+        self._attr_name = f"{name} {description.name}"
+
+        uuid = self.coordinator.data.get_uuid()
+        self._attr_unique_id = f"{uuid}-{description.key}"
+
+    @property
+    def is_on(self) -> bool | None:
+        """Return true if the binary sensor is on."""
+        return self.coordinator.data.get_metric(self.entity_description.metric_key) == 1
```

Comment:
get_metric is returning StateType. Will this always work, or additional type checks needed?

Reply:
In my understanding this will always work. There is no problem in comparing None or a random string with == 1 (it will just never match). There is also the alternative of doing if == 1 return True, if == 0 return False, else return None. I tested that as well and could not detect a difference in results (get_metric will never return anything other than 0 or 1, as far as I can tell by practical observation).

Reply:
This will always work indeed,  as the  expression results in a boolean type.